lib/efi.py

```diff
@@ -20,6 +22,31 @@
 import lib.commands as commands
 
 
+class _EfiGlobalTempdir:
+    _instance = None
+
+    def _safe_cleanup(self):
+        if self._instance is not None:
+            try:
+                self._instance.cleanup()
+            except OSError:
+                pass
+
+    def get(self):
+        if self._instance is None:
+            self._instance = TemporaryDirectory()
+            atexit.register(self._safe_cleanup)
+        return self._instance
+
+    def getfile(self, suffix=None, prefix=None):
+        fd, path = mkstemp(suffix=suffix, prefix=prefix, dir=self.get().name)
+        os.close(fd)
+        return path
+
+
+_tempdir = _EfiGlobalTempdir()
+
+
```

Comment:
What does this bring that a fixture could not achieve?

Reply:
`efi.py` is a standalone library that doesn't integrate with pytest at all, so fixtures can't be used here.